pkg/bbgo/order_executor_general.go

```diff
@@ -246,24 +245,34 @@ func (e *GeneralOrderExecutor) OpenPosition(ctx context.Context, options OpenPos
 		}
 	}
 
-	if options.MarketOrder {
-		submitOrder.Type = types.OrderTypeMarket
-	} else if options.LimitOrder {
+	if options.LimitOrder {
 		submitOrder.Type = types.OrderTypeLimit
 		submitOrder.Price = price
 	}
 
 	quantity := options.Quantity
 
+	market, ok := e.session.Market(e.symbol)
```

Comment:
the market is already defined e.position.Market

Reply:
done